frontend/apps/reader/modules/readerlink.lua

```diff
@@ -148,8 +160,97 @@ function ReaderLink:onGoBackLink()
 end
 
 function ReaderLink:onSwipe(_, ges)
-    if ges.direction == "east" and swipe_to_go_back() then
-        return self:onGoBackLink()
+    if ges.direction == "east" then
+        if isSwipeToGoBackEnabled() then
+            return self:onGoBackLink()
+        end
+    elseif ges.direction == "west" then
+        if isSwipeToFollowFirstLinkEnabled() then
+            return self:onGoToFirstLink(ges)
+        end
+    end
+end
+
+function ReaderLink:onGoToFirstLink(ges)
+    if not isFollowLinksOn() then return end
+    local firstlink = nil
+    if self.ui.document.info.has_pages then
+        local pos = self.view:screenToPageTransform(ges.pos)
+        if not pos then
+            return
+        end
+        local links = self.ui.document:getPageLinks(pos.page)
+        if #links == 0 then
+            return
+        end
+        -- DEBUG("PDF Page links : ", links)
+        -- We may get multiple links: internal ones (with "page" key)
+        -- that we're interested in, but also external links (no "page", but
+        -- a "uri" key) that we don't care about.
+        --     [2] = {
+        --         ["y1"] = 107.88977050781,
+        --         ["x1"] = 176.60360717773,
+        --         ["y0"] = 97.944396972656,
+        --         ["x0"] = 97,
+        --         ["page"] = 347
+        --     },
+        -- Links may not be in the order they are in the page, so let's
+        -- find the one with the smallest y0.
+        local first_y0 = nil
+        for _, link in ipairs(links) do
+            if link["page"] then
+                if first_y0 == nil or link["y0"] < first_y0 then
+                    -- onGotoLink()'s GotoPage event needs the link
+                    -- itself, and will use its "page" value
+                    firstlink = link
+                    first_y0 = link["y0"]
+                end
+            end
+        end
+    else
+        local links = self.ui.document:getPageLinks()
+        if #links == 0 then
+            return
```

Comment:
I haven't tested this on my dev machine yet. but based on your comment, this will skip the `clearSelection()` call and left the highlight on the screen?